src/libraries/ReservoirLibrary.sol

```diff
@@ -0,0 +1,224 @@
+        uint256 reserveOut,
+        uint256 swapFee
+    ) internal pure returns (uint256 amountOut) {
+        require(amountIn > 0, "RL: INSUFFICIENT_INPUT_AMOUNT");
+        require(reserveIn > 0 && reserveOut > 0, "RL: INSUFFICIENT_LIQUIDITY");
+        uint amountInWithFee = amountIn * (FEE_ACCURACY - swapFee);
+        uint numerator = amountInWithFee * reserveOut;
+        uint denominator = reserveIn * FEE_ACCURACY + amountInWithFee;
+        amountOut = numerator / denominator;
+    }
+
+    function getAmountInConstantProduct(
+        uint256 amountOut,
+        uint256 reserveIn,
+        uint256 reserveOut,
+        uint256 swapFee
+    ) internal pure returns (uint256 amountIn) {
+        require(amountOut > 0, "RL: INSUFFICIENT_OUTPUT_AMOUNT");
+        require(reserveIn > 0 && reserveOut > 0, "RL: INSUFFICIENT_LIQUIDITY");
+        uint numerator = reserveIn * amountOut * FEE_ACCURACY;
+        uint denominator = (reserveOut - amountOut) * (FEE_ACCURACY - swapFee);
+        amountIn = (numerator / denominator) + 1;
+    }
+
+    function getAmountOutStable(
+        uint256 amountIn,
+        uint256 reserveIn,
+        uint256 reserveOut,
+        uint256 swapFee,
+        ExtraData memory data
+    ) internal pure returns (uint256 amountOut) {
+        require(amountIn > 0, "RL: INSUFFICIENT_INPUT_AMOUNT");
+        require(reserveIn > 0 && reserveOut > 0, "RL: INSUFFICIENT_LIQUIDITY");
+
+        amountOut = StableMath._getAmountOut(
+            amountIn,
+            reserveIn,
+            reserveOut,
+            data.token0PrecisionMultiplier,
+            data.token1PrecisionMultiplier,
+            true,
+            swapFee,
+            2 * data.amplificationCoefficient
+        );
+    }
+
+    function getAmountInStable(
+        uint256 amountOut,
+        uint256 reserveIn,
+        uint256 reserveOut,
+        uint256 swapFee,
+        ExtraData memory data
+    ) internal pure returns (uint256 amountIn) {
+        require(amountOut > 0, "RL: INSUFFICIENT_OUTPUT_AMOUNT");
+        require(reserveIn > 0 && reserveOut > 0, "RL: INSUFFICIENT_LIQUIDITY");
+
+        amountIn = StableMath._getAmountIn(
+            amountOut,
+            reserveIn,
+            reserveOut,
+            data.token0PrecisionMultiplier,
+            data.token1PrecisionMultiplier,
+            false,
+            swapFee,
+            2 * data.amplificationCoefficient
+        );
+    }
+
+    // performs chained getAmountOut calculations on any number of pairs
+    function getAmountsOut(
+        address factory,
+        uint256 amountIn,
+        address[] calldata path,
+        uint256[] calldata curveIds
+    ) internal view returns (uint256[] memory amounts) {
+        require(path.length >= 2, "RL: INVALID_PATH");
+        require(curveIds.length == path.length - 1, "RL: CURVE_IDS_INVALID_LENGTH");
+        amounts = new uint[](path.length);
+        amounts[0] = amountIn;
+        for (uint i; i < path.length - 1; ) {
```

Comment:
`uint i = 0;` makes it more obvious contrast against the other function that does `uint i = path.length`, also `uint256` not `uint`

Reply:
fixed